### PR TITLE
doc: security: Add TF-M ITS encryption

### PR DESCRIPTION
**Pull request by mswarowsky**
Clone of the reviewed PR here: https://github.com/mswarowsky/sdk-nrf/pull/2 

Document how the nordic platforms provide the platform-specific elements of the ITS encryption of TF-M

### CODE REVIEW

**Review by SebastianBoe — APPROVED** [review]

File: doc/nrf/security/tfm.rst [hunk 1]
```diff
@@ -83,6 +83,32 @@ When :kconfig:option:`CONFIG_TFM_PROFILE_TYPE_MINIMAL` is set, the configurabili
...
+.. _tfm_encrypted_its:
+
+Encrypted ITS
+=============
+
+TF-M implements a PSA internal trusted storage (ITS) with encryption and authentication.
+For more information about the general features of the TF-M ITS service, see `TF-M ITS`_.
+
+To enable TF-M ITS encryption, use the Kconfig option :kconfig:option:`CONFIG_TFM_ITS_ENCRYPTED`.
+The ITS encryption is transparent to the user as long as the Master Key Encryption Key (MKEK) is populated before usage.
```
Comment (melwee01) [inline doc/nrf/security/tfm.rst:96]:
```suggestion
The ITS encryption is transparent to the user as long as the Master Key Encryption Key (MKEK) is populated before use.
```

File: doc/nrf/security/tfm.rst [hunk 2]
```diff
@@ -83,6 +83,32 @@ When :kconfig:option:`CONFIG_TFM_PROFILE_TYPE_MINIMAL` is set, the configurabili
...
+The MKEK is protected by the KMU peripheral and its key material cannot be read by software. It can only be used by reference.
+
+The file ID is used as a derivation label for the KDF.
+This means that each file ID uses a different AEAD key.
+As long as each file has a unique file ID, the key used for encryption and authentication is unique.
+
+To strengthen data integrity, the metadata of the ITS file (creation flags/size) is used as authenticated data in the encryption process.
+
+The nonce for the AEAD operation is generated by concatenating a random 8-byte seed and an increasing 4-byte counter.
+The random seed is generated once in the boot process and it stays the same until reset.
```
Comment (melwee01) [inline doc/nrf/security/tfm.rst:110]:
```suggestion
The random seed is generated once in the boot process and stays the same until reset.
```

**Review by melwee01 — APPROVED** [review]
Only two very small comments.

**Review by mia-ko — APPROVED** [review]
Should we add a line in release notes about this?

File: doc/nrf/security/tfm.rst [hunk 3]
```diff
@@ -83,6 +83,32 @@ When :kconfig:option:`CONFIG_TFM_PROFILE_TYPE_MINIMAL` is set, the configurabili
...
+=============
+
+TF-M implements a PSA internal trusted storage (ITS) with encryption and authentication.
+For more information about the general features of the TF-M ITS service, see `TF-M ITS`_.
+
+To enable TF-M ITS encryption, use the Kconfig option :kconfig:option:`CONFIG_TFM_ITS_ENCRYPTED`.
+The ITS encryption is transparent to the user as long as the Master Key Encryption Key (MKEK) is populated before usage.
+
+On Nordic Semiconductor devices, the hardware-accelerated AEAD scheme ChaChaPoly1305 is used with a 256 bits key.
+This key is derived with a key derivation function (KDF) based on CMAC.
```
Comment (Vge0rge) [inline doc/nrf/security/tfm.rst:99]: Probably better to mention NIST SP 800-108 here.

**Review by Vge0rge — APPROVED** [review]

**Comment by mswarowsky** [comment]
> Should we add a line in the release notes about this? I added a DNM label until we are ready.

The feature existed before, so I assumed we don't need it but sure I can add something.

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 4]
```diff
@@ -983,7 +983,9 @@ zcbor
 Trusted Firmware-M
 ==================
 
-|no_changes_yet_note|
+* Added:
+
+  * Internal Trusted Storage (ITS) with enrcyption, see :ref:`tfm_encrypted_its`
```
Comment (mia-ko) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:988]: You can consider saying 
* Section :ref:... describing Internal Trusted Storage (ITS) with encryption.

Comment (mswarowsky) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst]: Changed it

**Comment by mia-ko** [comment]
> @mia-ko is it now fine for you? Then feel free to remove the DNM flag and I can re-post a merge request

Looks ok to me :) I will remove the tag. But I see that the doc build is failing for some reason. Not sure why, don't see any doc errors.

**Pull request merged**